Mathlib/CategoryTheory/Sites/Coherent.lean

```diff
@@ -82,4 +83,148 @@ lemma isSheaf_coherent [Precoherent C] (P : Cᵒᵖ ⥤ Type w) :
     rintro B S ⟨α, _, X, π, rfl, hS⟩
     exact h _ _ _ _ hS
 
+
+namespace coherentTopology
+
+variable {C : Type _} [Category C] [Precoherent C]
+
+variable {X : C}
+/--
+For a precoherent category, any sieve that contains an `EffectiveEpiFamily` is a sieve of the
+coherent topology.
+Note: This is one direction of `mem_sieves_iff_hasEffectiveEpiFamily`, but is needed for the proof.
+-/
+theorem mem_sieves_of_hasEffectiveEpiFamily (S : Sieve X) :
+    (∃ (α : Type) (_ : Fintype α) (Y : α → C) (π : (a : α) → (Y a ⟶ X)),
+        EffectiveEpiFamily Y π ∧ (∀ a : α, (S.arrows) (π a)) ) →
+          (S ∈ GrothendieckTopology.sieves (coherentTopology C) X) := by
+  rintro ⟨α, ⟨h, ⟨Y, ⟨π, hπ⟩⟩⟩⟩
+  have h_le : Sieve.generate (Presieve.ofArrows _ π) ≤ S := by
+      rw [Sieve.sets_iff_generate (Presieve.ofArrows _ π) S]
+      apply Presieve.le_of_factorsThru_sieve (Presieve.ofArrows (fun i => Y i) π) S _
+      intro Y g f
+      use Y, 𝟙 Y
+      rcases f with ⟨i⟩
+      exact ⟨π i, ⟨hπ.2 i,Category.id_comp (π i) ⟩⟩
+  apply Coverage.saturate_of_superset (coherentCoverage C) h_le
+  exact Coverage.saturate.of X _ ⟨α, inferInstance, Y, π, ⟨rfl, hπ.1⟩⟩
+
+/-- Every Yoneda-presheaf is a sheaf for the coherent topology. -/
+theorem isSheaf_yoneda_obj (W : C) : Presieve.IsSheaf (coherentTopology C) (yoneda.obj W) := by
+  rw [isSheaf_coherent]
+  intro X α _ Y π H
+  have h_colim:= isColimitOfEffectiveEpiFamilyStruct Y π H.effectiveEpiFamily.some
+  rw [←Sieve.generateFamily_eq] at h_colim
+  intro x hx
+  let x_ext := Presieve.FamilyOfElements.sieveExtend x
+  have hx_ext := Presieve.FamilyOfElements.Compatible.sieveExtend hx
+  let S := Sieve.generate (Presieve.ofArrows Y π)
+  have := (Sieve.forallYonedaIsSheaf_iff_colimit S).mpr ⟨h_colim⟩ W x_ext hx_ext
+  rcases this with ⟨t, t_amalg, t_uniq⟩
+  use t
+  constructor
+  · convert Presieve.isAmalgamation_restrict (Sieve.le_generate (Presieve.ofArrows Y π)) _ _ t_amalg
+    refine Eq.symm (Presieve.restrict_extend hx)
```

Comment:
```suggestion
    exact (Presieve.restrict_extend hx).symm
```
No need to use refine if it's closing the goal, and dot notation makes things a bit more compact.